File: exercismdotio/python/grade-school/grade_school.py



class School():
	def __init__(self, name):
		self.name = name
		self.students = {} # keys are integer grades, values are lists of student names


	def grade(self, queryGrade):
		if queryGrade not in self.students:
			return []

		return self.students[queryGrade]


	def add(self, student, grade):
		self.students.setdefault(grade, [])

		self.students[grade].append(student)

	def sort(self):
		keys = list(self.students.keys())
		keys.sort()

		result = []
		for k in keys:
			studentNames = tuple(sorted(self.students[k]))
			result.append((k, studentNames))
		return result

File: exercismdotio/javascript/grade-school/grade-school.js


var School = function() {
    this.students = {};

};

School.prototype.add = function(studentName, gradeLevel) {
    if (this.students[gradeLevel] === undefined) {
        this.students[gradeLevel] = [];
    }

    this.students[gradeLevel].push(studentName);
    this.students[gradeLevel].sort();
};

School.prototype.roster = function() {
    return this.students;
};


School.prototype.grade = function(gradeLevel) {
    //return this.students[gradeLevel] || [];

    if (this.students[gradeLevel] === undefined) {
        return [];
    }
    else {
        return this.students[gradeLevel];
    }
};

module.exports = School;